File: EasePasswordDecrypt/EasePasswordDecrypt/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using System.Data.SqlClient;

namespace EasePasswordDecrypt
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            decrypt();
        }

        private void decrypt()
        {
            string password;
            string decrypted = "";
            password = textBox1.Text;

            for (int i = 0; i < password.Length; i++)
            {
                decrypted = decrypted + Strings.Chr(Strings.Asc(password.Substring(i, 1)) - (i + 1));
            }

            txtDecrypted.Text = decrypted;
        }

        private void PopulateGrid()
        {
            string serverName;
            string userName;
            string password;

            serverName = txtServerName.Text;
            userName = txtUserName.Text;
            password = txtPassword.Text;

            string constring = @"Server="+serverName+"; Database=AML_PROD;User="+userName+";Password="+password+";";
            using (SqlConnection con = new SqlConnection(constring))
            {
                using (SqlCommand cmd = new SqlCommand("Select userid, username, passwordx From euser", con))
                {
                    cmd.CommandType = CommandType.Text;
                    using (SqlDataAdapter sda = new SqlDataAdapter(cmd))
                    {
                        using (DataTable dt = new DataTable())
                        {


                            sda.Fill(dt);
                            dataGridView1.DataSource = dt;
                        }
                    }
                }
            }
        }

        private void button2_Click(object sender, EventArgs e)
        {
            PopulateGrid();
        }
    }
}

File: EasePasswordDecryptVB/EasePasswordDecryptVB/Form1.vb
Imports System.Data.SqlClient
Public Class Form1

    Dim ServerName
    Dim UserName
    Dim Password

    Public Sub btnDecrypt_Click(sender As Object, e As EventArgs) Handles btnDecrypt.Click
        Dim password As String
        Dim decrypt As String

        password = txtEasePassword.Text

        decrypt = Getdecrypt(password)

        txtDecrypted.Text = decrypt

    End Sub

    Public Sub btnConnect_Click(sender As Object, e As EventArgs) Handles btnConnect.Click


        ServerName = txtServerName.Text
        UserName = txtUserName.Text
        Password = txtPassword.Text



        Dim con As New SqlConnection()
        con.ConnectionString = "Data Source=" & ServerName & ";uid=" & UserName & ";pwd=" & Password & ";database=aml_prod"
        con.Open()
        Dim da As New SqlDataAdapter("Select * From sys.databases order by name", con)
        Dim table As New DataTable()
        da.Fill(table)
        cmb_Databases.DataSource = New BindingSource(table, Nothing)
        cmb_Databases.DisplayMember = "Name"
        cmb_Databases.ValueMember = "database_id"

    End Sub

    Public Sub cmb_Databases_SelectionChange(ByVal sender As Object, ByVal e As System.EventArgs) Handles cmb_Databases.SelectionChangeCommitted
        Dim selectedItem As Object
        selectedItem = cmb_Databases.GetItemText(cmb_Databases.SelectedItem)

        'MessageBox.Show("Selected Item Text: " & selectedItem.ToString())

        Dim str As String = "Data Source=" & ServerName & ";uid=" & UserName & ";pwd=" & Password & ";database=aml_prod"
        Dim con As New SqlConnection(str)
        Dim com1 As String = "Select userid, username, passwordx From " & selectedItem.ToString() & ".dbo.euser"

        Dim Adpt As New SqlDataAdapter(com1, con)
        Dim ds As New DataSet()
        Adpt.Fill(ds, "Emp")
        DataGridView1.DataSource = ds.Tables(0)

    End Sub

    Public Function Getdecrypt(ByVal password) As String

        Dim decrypted As String = "", intI As Int16 = 0

        Try
            For intI = 1 To Len(password)
                decrypted &= Chr(Asc(Mid(password, intI, 1)) - intI)
            Next
        Catch ex As Exception
            MessageBox.Show("Failed to Decrypt")
        End Try

        Return decrypted

    End Function



End Class
